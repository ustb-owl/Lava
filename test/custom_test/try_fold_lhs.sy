int f() {
  return 233;
}

int g1() {
  int a = f();
  int b = a + 456;
  int c = b + 123;
  return c;
}

int g2() {
  int a = f();
  int b = a + 456;
  int c = b - 123;
  return c;
}

int g3() {
  int a = f();
  int b = a + 456;
  int c = 123 - b;
  return c;
}

int g4() {
  int a = f();
  int b = a - 456;
  int c = b + 123;
  return c;
}

int g5() {
  int a = f();
  int b = a - 456;
  int c = b - 123;
  return c;
}

int g6() {
  int a = f();
  int b = a - 456;
  int c = 123 - b;
  return c;
}

int g7() {
  int a = f();
  int b = 456 - a;
  int c = b + 123;
  return c;
}

int g8() {
  int a = f();
  int b = 456 - a;
  int c = b - 123;
  return c;
}

int g9() {
  int a = f();
  int b = 456 - a;
  int c = 123 - b;
  return c;
}

int main() {
  putint(g1()); putch(10);
  putint(g2()); putch(10);
  putint(g3()); putch(10);
  putint(g4()); putch(10);
  putint(g5()); putch(10);
  putint(g6()); putch(10);
  putint(g7()); putch(10);
  putint(g8()); putch(10);
  putint(g9()); putch(10);
}