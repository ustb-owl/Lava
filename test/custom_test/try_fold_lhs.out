812
566
-566
-100
-346
346
346
100
-100
0
